03_letters/mohitnag/main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLetters(t *testing.T) {
+	testData := []struct {
+		Scenario string
+		input    string
+		expected map[rune]int
+	}{
+		{"Scenario One", "eaa!baabq23123!", map[int32]int{33: 2, 49: 1, 50: 2, 51: 2, 97: 4, 98: 2, 101: 1, 113: 1}},
```

Comment:
I like the way you have structured your test cases here, but we can do a couple of things to improve this further.

1. You have stated the expected variable is of type `map[rune]int` however when assigning values for this you are using a `map[int32]int`. This makes things a lot harder to interpret as you are assigning the int representation of the rune. Can you change this so your keys to the expected map are actual rune characters?

2. Your `Scenario` name could be a bit more descriptive. The names `"Scenario One"` (and two) don't give much information about what it is testing. These should be something more like, "Empty String" etc which describes the input under test.

3. You could add a few more test cases as well as the empty and non empty cases. Since we are using runes, why don't you add a case which involves special characters, and emoji unicode characters?

Reply:
Agree. Done the changes

Reply:
With number one this is still not fully complete. You have updated this to be of type rune, but you should be using the actual characters which it refers to rather than the int representation. This improves readability.